lib/compat.php

```diff
@@ -175,3 +175,36 @@ function gutenberg_get_post_from_context() {
 	}
 	return get_post();
 }
+
+/**
+ * Shim that hooks into `pre_render_block` so as to override `render_block` with
+ * a function that assigns block context.
+ *
+ * This can be removed when plugin support requires WordPress 5.5.0+.
+ *
+ * @see https://core.trac.wordpress.org/ticket/49927
+ *
+ * @param string|null $pre_render   The pre-rendered content. Defaults to null.
+ * @param array       $parsed_block The parsed block being rendered.
+ *
+ * @return string String of rendered HTML.
+ */
+function gutenberg_render_block_with_assigned_block_context( $pre_render, $parsed_block ) {
+	global $post;
+
+	// If a non-null value is provided, a filter has run at an earlier priority
+	// and has already handled custom rendering and should take precedence.
+	if ( null !== $pre_render ) {
+		return $pre_render;
+	}
+
+	$source_block = $parsed_block;
+
+	/** This filter is documented in src/wp-includes/blocks.php */
+	$parsed_block = apply_filters( 'render_block_data', $parsed_block, $source_block );
+	$context      = array( 'postId' => $post->ID );
```

Comment:
We need `postType` too.

Reply:
> We need `postType` too.

Practically speaking, maybe not? `postId` are globally unique, and one can derive the post's details (including its type) from the ID alone when server-side. This is unlike the browser, where in order to interact with the REST API to retrieve post details, [one must know both the ID and the type](https://wordpress.slack.com/archives/C02RQC26G/p1546983215113700). At least, that was my original thinking with how it was implemented here.

But since the block manifest is expected to be the same between the server and browser, and since it will most likely include both `postId` and `postType`, I guess it should be included, even if in most cases it won't really be needed on the server. It'd also be a good consistency to have between the two environments.

Reply:
> We need `postType` too.

Added `postType` in cd3d81b07.